### PR TITLE
fix: curio: rename lotus-provider to curio

### PR DESCRIPTION
**Pull request by LexLuthr**
## Proposed Changes

Rename `lotus-provider` to `curio`.

## Additional Info

The original source code directory `provider` has been renamed to `curiosrc`. I would have like to keep it `curio` but that creates a conflict with the binary of same name. Thus the name `curiosrc`. 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by LexLuthr** [comment]
@snadrus I was getting some warning like below for translations. I am assuming these are because of the missing tranlation values. Or maybe I did something wrong while generating them. Either way, we need to add a `.doc` file for translation process to make it easier for other to contribute.

```
ko: Missing entry for "This interactive tool will walk you through migration of curio.\nPress Ctrl+C to exit at any time.".
ko: Missing entry for "This tool confirms each action it does and each step is reversable.".
```

**Comment by snadrus** [comment]
Please see the top of the file for translation directions.

File: node/repo/fsrepo.go [hunk 1]
```diff
@@ -185,27 +185,27 @@ func (worker) APIInfoEnvVars() (primary string, fallbacks []string, deprecated [
...
 
-func (provider) RepoFlags() []string {
-	return []string{"provider-repo"}
+func (curio) RepoFlags() []string {
+	return []string{"curio-repo"}
 }
 
-func (provider) APIInfoEnvVars() (primary string, fallbacks []string, deprecated []string) {
+func (curio) APIInfoEnvVars() (primary string, fallbacks []string, deprecated []string) {
 	return "PROVIDER_API_INFO", nil, nil
```
Comment (snadrus) [inline node/repo/fsrepo.go:209]: oops?

**Review by snadrus — CHANGES_REQUESTED** [review]

File: node/repo/fsrepo.go (hunk 1 above)
Comment (LexLuthr) [reply node/repo/fsrepo.go:209]: Yep!! Let me recheck other once more. I will add a new commit.

**Comment by snadrus** [comment]
@LexLuthr Please resolve conflicts & submit.

**Review by snadrus — APPROVED** [review]

File: node/repo/fsrepo.go (hunk 1 above)
Comment (LexLuthr) [reply node/repo/fsrepo.go]: Fixed the API_INFO. That was the only mistake. We should probably rebase `easyMigration` on master as Magik has pushed multiple new change with provider name.

**Pull request merged**